src/promptflow/promptflow/batch/_python_executor_proxy.py

```diff
@@ -22,6 +22,33 @@ class PythonExecutorProxy(AbstractExecutorProxy):
     def __init__(self, flow_executor: FlowExecutor):
         self._flow_executor = flow_executor
 
+    @classmethod
+    def _generate_metadata(cls, flow_file: Path, working_dir: Path):
```

Comment:
@D-W- dump of the generated metadata looks required for csharp (and other potential new languages) given they can't do reflection within current process. So do you think it's okay to always dump them? or we still skip dump for python if possible, like local `pf flow test`?